### PR TITLE
python3Packages.tensorflow-metadata: init at 1.5.0

### PR DESCRIPTION
**Pull request by ndl**
###### Motivation for this change

This is one of the prerequisites for `tensorflow-datasets`.

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.05 Release Notes (or backporting 21.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2205-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by ndl** [comment]
@samuela This is **not** a required dependency for JAX libs PRs so not assigning you as a reviewer (you had reviewed already quite a lot of PRs from me recently - thanks a lot for that, BTW!), but just in case you're interested / have free cycles - feel free to have a look :-)

The idea is to get `tensorflow-datasets` integrated as it provides the access to multiple datasets, but there are a couple of missing dependencies in `nixpkgs` ATM, `tensorflow-metadata` is one of them.

**Comment by samuela** [comment]
this is great! packaging tensorflow-datasets has been a longstanding goal of mine

**Comment by samuela** [comment]
Result of `nixpkgs-review pr 153767` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>2 packages failed to build:</summary>
  <ul>
    <li>python38Packages.tensorflow-metadata</li>
    <li>python39Packages.tensorflow-metadata</li>
  </ul>
</details>

**Comment by samuela** [comment]
Hmm, for some reason nixpkgs-review fails to build the packages, but if I manually `nix-build` them it works. Anyone know why that might be?

**Comment by ndl** [comment]
> Hmm, for some reason nixpkgs-review fails to build the packages, but if I manually nix-build them it works. 

Apparently the alias `googleapis_common_protos` behaved kind of strange, switching to `googleapis-common-protos` allows the checks to pass on Darwin + seems to fix `nixpkgs-review` build problem (at least it succeeds now for me locally). Please try again?

**Comment by samuela** [comment]
Result of `nixpkgs-review pr 153767` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>2 packages built:</summary>
  <ul>
    <li>python38Packages.tensorflow-metadata</li>
    <li>python39Packages.tensorflow-metadata</li>
  </ul>
</details>

File: pkgs/development/python-modules/tensorflow-metadata/default.nix [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+  };
+
+  patches = [
+    ./build.patch
+  ];
+
+  # Default build pulls in Bazel + extra deps, given the actual build
+  # is literally three lines (see below) - replace it with custom build.
+  preBuild = ''
+    set -e
```
Comment (samuela) [inline pkgs/development/python-modules/tensorflow-metadata/default.nix:26]: Is `set -e` not the default in nix shell stuff already?

**Review by samuela — COMMENTED** [review]
LGTM except for one question about the `set -e` thing

File: pkgs/development/python-modules/tensorflow-metadata/default.nix (hunk 1 above)
Comment (ndl) [reply pkgs/development/python-modules/tensorflow-metadata/default.nix]: Apparently yes (adding failing command aborts the build indeed even without this setting), thanks for noticing! Removed. Was confused by seeing it regularly in other derivations - looks like this is not a common knowledge?

Comment (samuela) [reply pkgs/development/python-modules/tensorflow-metadata/default.nix]: Yeah, that's a good question... I would certainly hope that `set -e` is the default but perhaps it hasn't always been that way? I have no idea...

**Review by samuela — APPROVED** [review]
LGTM I'll squash and merge tomorrow unless someone objects

**Pull request merged**